### DISCUSSION TITLE
Unable to import module 'testdb': No module named 'psycopg2._psycopg'

### DISCUSSION THREAD

**Discussion by bkrishnaiah** [aws/aws-cdk#28339] [Q&A]
Hi,

I am trying to create a lambda function where I can have Postgres DB connection with AWS RDS.  I am using the tech stack as python and aws cdk.
I have installed psycopg2-binary library to enable the DB connection. I have deployed this lambda function to AWS environment. But while testing the lambda in AWS console, I am getting below error. Could someone help me on this. 

For reference, 
————————
> pip install -t "my_dir" "psycopg2-binary"

> my “test_db.py” looks like below and the db connection values used as per my requirement.

import psycopg2

connection = psycopg2.connect(
    host=db_host,
    user=user_name,
    password=user_password,
    database=dbname
)

Error Message:
—————————
{
  "errorMessage": "Unable to import module 'testdb': No module named 'psycopg2._psycopg'",
  "errorType": "Runtime.ImportModuleError",
  "requestId": "593f9067-he23-4a54-b7773-39acdb2cd723",
  "stackTrace": []
}

Thanks in advance..!!!

**Comment by tomoish** [comment]
Hello, @bkrishnaiah.
There is a [psycopg2 Python Library for AWS Lambda](https://github.com/jkehler/awslambda-psycopg2) , which could be a viable solution for your question.

I'm not sure about your Python version. If this tool doesn't support your version, using Docker would be a good alternative.

Below is a solution using Docker (in this example, I use Python 3.9):

```
# in your working directory
$ echo 'psycopg2-binary' > requirements.txt
# run docker and run `pip install`
$ docker run -v "$PWD":/var/task "amazon/aws-sam-cli-build-image-python3.9" /bin/sh -c "pip install -r requirements.txt -t python/lib/python3.9/site-packages/; exit"
# zip psycopg2 library
$ zip -r psycogpg2.zip python
```
